sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -501,32 +501,44 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
   ): Promise<UpdateResult> {
     const auditUser = { id: auditUserId } as User;
     const now = new Date();
-    const updateResult = await this.repo
-      .createQueryBuilder()
-      .update(DisbursementSchedule)
-      .set({
-        coeStatus: COEStatus.declined,
-        coeUpdatedBy: auditUser,
-        coeUpdatedAt: now,
-        coeDeniedReason: { id: coeDeniedReasonId },
-        coeDeniedOtherDesc:
-          coeDeniedReasonId === COE_DENIED_REASON_OTHER_ID
-            ? otherReasonDesc
-            : null,
-        modifier: auditUser,
-        updatedAt: now,
-      })
-      .where("id = :disbursementScheduleId", { disbursementScheduleId })
-      .andWhere("coeStatus = :required", { required: COEStatus.required })
-      .execute();
+    return await this.dataSource.transaction(
```

Comment:
await?

Reply:
@ann-aot is the suggestion to remove the `await` here?

Reply:
do we await here?